### ISSUE TITLE
Ability to customize table row limit in subscriptions

### ISSUE THREAD

**Issue by ignacio-mb** [metabase/metabase#32165]
**Is your feature request related to a problem? Please describe.**
When a table has 11 rows and you can't see the full table in an email, and you have to open the csv to see the remaining row, that can be a bit annoying. Requested by a customer as well.

**Describe the solution you'd like**
Ability to change the number of rows an email can render from the subscription setting. There may be an impossibility that I am not aware of.
There should be a limit that Metabase has to impose though, like 20 or something, but more than 10 can be very useful.

**Describe alternatives you've considered**
NA

**How important is this feature to you?**
Requested by a customer that sends a lot of emails, that at the same time has customers that send a lot of emails and encounter with this situation. 

**Additional context**
NA

**Comment by janmechtel** [comment]
This seems to not only impact subscriptions but also viewing dashboard on mobile will limit the row size :-(

**Comment by janmechtel** [comment]
I started investigating this more, I'm a beginner in react. Would be great to understand what the requirements would be to submit a pull request for this.

I'd imagine there could be a setting "row limit on small screens" which seems to be 5 rows at the moment, and users could set it to 0 to indicate "keep original height" as specified in the full screen dashboard.

As a hacky workaround I found that making the table much more narrow (= mobile friendly) but keeping the dashboard in full screen, will keep the height and avoid pagination. So I'll take a screenshot with a wide screen (1000px) and then cut away the 500px that I don't need, keeping only the narrow table in the 500px.  
![image](https://github.com/metabase/metabase/assets/1004314/a569d0fc-af46-43ef-b0eb-96cb0de60ebc)
